Sources/BraintreeCore/BTPaymentMethodNonceParser.swift

```diff
@@ -53,18 +53,80 @@ import Foundation
             return nil
         }
 
-        if let completionHandler = completionHandler {
+        if let completionHandler {
             return completionHandler(json)
         }
 
-        if json?["nonce"].isString != false {
+        if json?["nonce"].isString == false {
+            return nil
+        }
+
+        let type = json?["type"].asString()
+
+        if type == "CreditCard", let cardType = json?["details"]["cardType"].asString() {
```

Comment:
The below logic mirrors the behavior of iOS v5 with regards to the type. Though this is with the constraints of not being able to cast to unknown nonce types in Swift. The load method used previously was a very hacky workaround and does not map to Swift - the method (`load`) in Obj-C is prohibited in Swift. Currently the JS SDK has the method `fetchPaymentMethodNonces` in Core while Android has this method in the Drop-in. We should consider where this method should live and align across all 3 platforms.